aspnetcore/fundamentals/startup.md

```diff
@@ -38,35 +38,38 @@ In the following example from the default web site template, several extension m
 
 For features that require substantial setup there are `Add[Service]` extension methods on [IServiceCollection](https://docs.microsoft.com/en-us/aspnet/core/api/microsoft.extensions.dependencyinjection.iservicecollection). This example from the default web site template configures the app to use services for Entity Framework, Identity, and MVC:
 
 [!code-csharp[Main](../common/samples/WebApplication1/Startup.cs?highlight=4,7,11&start=40&end=55)]
 
-Adding services to the services container makes them available within your application via [dependency injection](dependency-injection.md).
+Adding services to the services container makes them available within your application via [dependency injection](xref:fundamentals/dependency-injection).
 
 ## Services Available in Startup
 
-ASP.NET Core dependency injection provides application services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or one of its `Configure` or `ConfigureServices` methods. 
+ASP.NET Core dependency injection provides application services during an application's startup. You can request these services by including the appropriate interface as a parameter on your `Startup` class's constructor or its `Configure` method. The `ConfigureServices` method only takes an `IServiceCollection` parameter (but any registered service can be retrieved from this collection, so additional parameters are not necessary).
 
-Looking at each method in the `Startup` class in the order in which they are called, the following services may be requested as parameters:
+Below are some of the services typically requested by `Startup` methods:
 
-* In the constructor:  `IHostingEnvironment`, `ILoggerFactory`
-* In the `ConfigureServices` method:  `IServiceCollection`
-* In the `Configure` method:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`, `IApplicationLifetime`
+* In the constructor:  `IHostingEnvironment`, `ILogger<Startup>`
+* In `ConfigureServices`:  `IServiceCollection`
+* In `Configure`:  `IApplicationBuilder`, `IHostingEnvironment`, `ILoggerFactory`
+
+Any services added by the ``WebHostBuilder`` ``ConfigureServices`` method may be requested by the ``Startup`` class constructor or its ``Configure`` method. Use `WebHostBuilder` to provide any services you need during `Startup` methods.
 
 ## Additional Resources
 
-* [Working with Multiple Environments](environments.md)
-* [Middleware](middleware.md)
-* [Logging](logging.md)
-* [Configuration](configuration.md)
+* [Working with Multiple Environments](xref:fundamentals/environments)
+* [Middleware](xref:fundamentals/middleware)
+* [Logging](xref:fundamentals/logging)
+* [Configuration](xref:fundamentals/configuration)
+* [List All Services Available to an ASP.NET Core App](https://github.com/ardalis/AspNetCoreStartupServices)
```

Comment:
@danroth27  https://github.com/ardalis/AspNetCoreStartupServices is a nice sample and useful for many folks. 
We will be creating a master repro for docs samples (every sample under the master will be it's own repo - like https://github.com/Azure-Samples
We could move the sample to our docs sample repro.